h2o-py/h2o/model/model_base.py

```diff
@@ -1506,193 +1409,28 @@ def varimp_plot(self, num_of_features=None, server=False):
-        if num_of_features is None:
-            num_of_features = min(len(val), 10)
-
-        fig, ax = plt.subplots(1, 1, figsize=(14, 10))
-        # create separate plot for the case where num_of_features == 1
-        if num_of_features == 1:
-            plt.barh(pos[0:num_of_features], val[0:num_of_features], align="center",
-                     height=0.8, color="#1F77B4", edgecolor="none")
-            # Hide the right and top spines, color others grey
-            ax.spines["right"].set_visible(False)
-            ax.spines["top"].set_visible(False)
-            ax.spines["bottom"].set_color("#7B7B7B")
-            ax.spines["left"].set_color("#7B7B7B")
-            # Only show ticks on the left and bottom spines
-            ax.yaxis.set_ticks_position("left")
-            ax.xaxis.set_ticks_position("bottom")
-            plt.yticks(pos[0:num_of_features], feature_labels[0:num_of_features])
-            ax.margins(None, 0.5)
-
-        else:
-            plt.barh(pos[0:num_of_features], val[0:num_of_features], align="center",
-                     height=0.8, color="#1F77B4", edgecolor="none")
-            # Hide the right and top spines, color others grey
-            ax.spines["right"].set_visible(False)
-            ax.spines["top"].set_visible(False)
-            ax.spines["bottom"].set_color("#7B7B7B")
-            ax.spines["left"].set_color("#7B7B7B")
-            # Only show ticks on the left and bottom spines
-            ax.yaxis.set_ticks_position("left")
-            ax.xaxis.set_ticks_position("bottom")
-            plt.yticks(pos[0:num_of_features], feature_labels[0:num_of_features])
-            plt.ylim([min(pos[0:num_of_features])- 1, max(pos[0:num_of_features])+1])
-            # ax.margins(y=0.5)
-
-        # check which algorithm was used to select right plot title
-        if self._model_json["algo"] == "gbm":
-            plt.title("Variable Importance: H2O GBM", fontsize=20)
-            if not server: plt.show()
-        elif self._model_json["algo"] == "drf":
-            plt.title("Variable Importance: H2O DRF", fontsize=20)
-            if not server: plt.show()
-        elif self._model_json["algo"] == "xgboost":
-            plt.title("Variable Importance: H2O XGBoost", fontsize=20)
-            if not server: plt.show()
-        # if H2ODeepLearningEstimator has variable_importances == True
-        elif self._model_json["algo"] == "deeplearning":
-            plt.title("Variable Importance: H2O Deep Learning", fontsize=20)
-            if not server: plt.show()
-        elif self._model_json["algo"] == "glm":
-            plt.title("Variable Importance: H2O GLM", fontsize=20)
-            if not server: 
-                plt.show()            
+        # For now, redirect to h2o.model.extensions.varimp for models that support the feature, and raise legacy error for others.
+        # Later, the method will be exposed only for models supporting the feature.
```

Comment:
@michalkurka as discussed, keeping the all those features exposed in `ModelBase` for now.
Second step (exposing features only for relevant models), will just be a matter of deleting the `ModelBase` methods and removing the leading underscore to make the implementation "public" in the feature mixin classes.